jerry-core/jerry.c

```diff
@@ -2361,7 +2372,14 @@ jerry_exec_snapshot (const void *snapshot_p, /**< snapshot */
   {
     vm_init (bytecode_p, false);
 
-    ret_code = vm_run_global ();
+    ecma_object_t *error_obj_p = NULL;
+
+    ret_code = vm_run_global (&error_obj_p);
+
+    if (ret_code == JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION)
+    {
+      ecma_deref_object (error_obj_p);
+    }
```

Comment:
There should be an ASSERT here that error_obj_p is NULL or not-NULL depending on the ret_code